lib/_tls_wrap.js

```diff
@@ -374,6 +375,12 @@ TLSSocket.prototype._wrapHandle = function(wrap) {
     res = null;
   });
 
+  if (wrap instanceof Stream) {
```

Comment:
Would it make sense if the `StreamBase` destructor had a way of telling the `TLSWrap` that it is getting destroyed? That certainly adds a bit of complexity but would also seem more reliable/“obviously correct” to me… @indutny ?